dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+            .append(" data-dot-inode=")
+            .append("\"" + content.getInode() + "\"")
+            .append(" data-dot-identifier=")
+            .append("\"" + content.getIdentifier() + "\"")
+            .append(" data-dot-type=")
+            .append("\"" + content.getContentType()
+                .variable() + "\"")
+            .append(" data-dot-lang=")
+            .append("\"" + content.getLanguageId() + "\"")
+            .append(">");
+
+
+        sb.append("#if($API_EDIT_MODE)")
+            .append(contentDiv)
+            .append("#end");
+
+
+
+        // CONTENTLET CONTROLS BEGIN
+        sb.append("#if($EDIT_MODE)");
+        sb.append("#set( $EDIT_CONTENT_PERMISSION=$EDIT_CONTENT_PERMISSION")
+            .append(identifier.getId())
+            .append(")");
+        sb.append("#end");
+
+        sb.append("#set($CONTENT_INODE='")
+            .append(content.getInode())
+            .append("' )")
+            .append("#set($IDENTIFIER_INODE='")
+            .append(identifier.getId())
+            .append("' )");
+
+        // set all properties from the contentlet
+        sb.append("#set($ContentInode='")
+            .append(content.getInode())
+            .append("' )")
+            .append("#set($ContentIdentifier='")
+            .append(identifier.getId())
+            .append("' )")
+            .append("#set($ContentletTitle=\"")
+            .append(UtilMethods.espaceForVelocity(conAPI.getName(content, APILocator.getUserAPI()
+                .getSystemUser(), true)))
+            .append("\" )");
+        String modDateStr = UtilMethods.dateToHTMLDate((Date) content.getModDate(), "yyyy-MM-dd H:mm:ss");
+        sb.append("#set($ContentLastModDate= $date.toDate(\"yyyy-MM-dd H:mm:ss\", \"")
+            .append(modDateStr)
+            .append("\"))")
+            .append("#set($ContentLastModUserId= \"")
+            .append(content.getModUser())
+            .append("\")");
+        if (content.getOwner() != null)
+            sb.append("#set($ContentOwnerId= \"")
+                .append(content.getOwner())
+                .append("\")");
+
+        // Structure fields
+
+        ContentType type = content.getContentType();
+
+        List<Field> fields = type.fields();
+
+
+        String widgetCode = "";
+
+        for (Field field : fields) {
+
+            String contField = field.dbColumn();
+            String contFieldValue = null;
+            Object contFieldValueObject = null;
+            String velPath = (!EDIT_MODE) ? "live/" : "working/";
+            if (field instanceof HiddenField || field instanceof ConstantField) {
+                if (field.variable()
+                    .equals("widgetPreexecute")) {
+                    continue;
+                }
+                if (field.variable()
+                    .equals("widgetCode")) {
+                    widgetCode = "#set($" + field.variable() + "=$velutil.mergeTemplate(\"" + velPath + content.getInode()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "#set($" 36 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)